packages/stateManager/index.ts

```diff
@@ -615,8 +628,41 @@ export class StateManager {
       this.keycards,
     ];
     const stream = this.client.createEventStream();
+    const shopClient = createPublicClient({
+      chain: process.env.DEV ? hardhat : mainnet,
+      transport: http(),
+    });
```

Comment:
```suggestion
```
this should just be an argument in constructor. The reason is the most likely the frontend need to create a publicClient anyways (https://github.com/masslbs/Tennessine/blob/90c10f0f0baa6c0ae5a01283ea4a26741ce3f437/packages/frontend/src/context/MyContext.tsx#L127) and we will want to use that one instead of creating two instances